autopilot/autopilot.go

```diff
@@ -8,24 +8,53 @@ import (
 	"go.sia.tech/renterd/bus"
 	"go.sia.tech/renterd/hostdb"
 	"go.sia.tech/renterd/internal/consensus"
+	rhpv2 "go.sia.tech/renterd/rhp/v2"
+	"go.sia.tech/renterd/wallet"
 	"go.sia.tech/renterd/worker"
+	"go.sia.tech/siad/types"
 )
 
 type Store interface {
 	Tip() consensus.ChainIndex
+	Synced() bool // TODO: should be added to `Tip` | should be moved over to Bus
+
 	Config() Config
 	SetConfig(c Config) error
 }
 
+// TODO: should be defined in bus package, should be the interface of the client
 type Bus interface {
+	// wallet
+	WalletBalance() (types.Currency, error)
+	WalletAddress() (types.UnlockHash, error)
+	WalletTransactions(since time.Time, max int) ([]wallet.Transaction, error)
+	WalletFund(txn *types.Transaction, amount types.Currency) ([]types.OutputID, []types.Transaction, error)
+	WalletDiscard(txn types.Transaction) error
+	WalletSign(txn *types.Transaction, toSign []types.OutputID, cf types.CoveredFields) error
+
+	// hostdb
 	AllHosts() ([]hostdb.Host, error)
 	Hosts(notSince time.Time, max int) ([]hostdb.Host, error)
+	Host(hostKey consensus.PublicKey) (hostdb.Host, error)
 	RecordHostInteraction(hostKey consensus.PublicKey, hi hostdb.Interaction) error
+
+	// contracts
+	AddContract(c rhpv2.Contract) error
+	RenewableContracts(renewWindow uint64) ([]bus.Contract, error)
```

Comment:
I'd make this more generic to avoid having a `renewWindow` in the store. Maybe pass in a block height instead of the renew window and return all contracts that have a block height smaller than that.